internal/proxy/roundtripper.go

```diff
@@ -29,16 +29,18 @@ type h2cTransportWrapper struct {
 }
 
 func (t *h2cTransportWrapper) RoundTrip(req *http.Request) (*http.Response, error) {
+	logger := log.ZapFromContext(req.Context()).GetInternalZapLogger().Desugar().With(
+		zap.String("origin", "RoundTrip"),
```

Comment:
why is origin RoundTrip here?

Reply:
origin is from where log was exected

Reply:
oh okay. i would suggest to look at how to configure automatic inclusion of source information in Zap. it will automatically add the go file name and line_no, function name, etc. to each log entry.

issue with this approach is, if someone changes the function name, it's not guaranteed that they will change the log field as well. once they drift, this log is going to cause more confusion.

Reply:
ok, makes sense